### DISCUSSION TITLE
Compatibility with uWebsockets.js

### DISCUSSION THREAD

**Discussion by Numer0us** [TooTallNate/proxy-agents#276] [Q&A]
I apologize for the general question. I’m a fairly new developer just looking for a little guidance.

Can the http proxy agent be used to establish a websocket connection with the [uWebsockets.js](https://github.com/uNetworking/uWebSockets.js/) library through an http proxy server? Is it only compatible with ws?

**Comment by TooTallNate** [comment]
From what I can tell, [uWebsockets.js](https://github.com/uNetworking/uWebSockets.js/) is for writing WebSocket _servers_. These proxy agent packages are for making _client_ requests. That said, these packages are known working and tested against the popular [`ws`](https://www.npmjs.com/package/ws) package, which offers a WebSocket client.

**Reply by Numer0us** [reply]
Ahh ok that makes sense. Thanks for the reply. 

They work quite well with ws. Thanks for maintaining.
